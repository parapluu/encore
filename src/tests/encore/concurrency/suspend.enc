active class Main
  def foo() : int
    0
  end

  def main() : unit
    let
      t = new Test
    in
        t ! fst(this)
        t ! snd()
    end
  end
end
active class Test
  def fst(m:Main) : unit
    print("Before suspend\n")
    get(m ! foo())
    this.suspend()
    print("After suspend\n")
  end
  def snd() : unit
    print("Interleaving!\n")
  end
end
