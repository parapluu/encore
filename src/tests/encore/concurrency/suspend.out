Before suspend
Interleaving!
After suspend
